crates/router/src/core/payments/operations/payment_session.rs

```diff
@@ -299,84 +309,88 @@ where
             .change_context(errors::ApiErrorResponse::InternalServerError)
             .attach_printable("Database error when querying for merchant connector accounts")?;
 
+        let filtered_connector_accounts = helpers::filter_mca_based_on_business_details(
+            all_connector_accounts,
+            Some(payment_intent),
+        );
+
+        let requested_payment_method_types = request.wallets.clone();
         let mut connector_and_supporting_payment_method_type = Vec::new();
 
-        for connector_account in connector_accounts {
-            let payment_methods = connector_account
-                .payment_methods_enabled
-                .unwrap_or_default();
-            for payment_method in payment_methods {
-                let parsed_payment_method_result: Result<
-                    PaymentMethodsEnabled,
-                    error_stack::Report<errors::ParsingError>,
-                > = payment_method.clone().parse_value("payment_method");
-
-                match parsed_payment_method_result {
-                    Ok(parsed_payment_method) => {
-                        let payment_method_types = parsed_payment_method
+        filtered_connector_accounts
+            .into_iter()
+            .for_each(|connector_account| {
+                let res = connector_account
+                    .payment_methods_enabled
+                    .unwrap_or_default()
+                    .into_iter()
+                    .map(|payment_methods_enabled| {
+                        payment_methods_enabled
+                            .parse_value::<PaymentMethodsEnabled>("payment_methods_enabled")
+                    })
+                    .filter_map(|parsed_payment_method_result| {
+                        let error = parsed_payment_method_result.as_ref().err();
+                        logger::error!(session_token_parsing_error=?error);
```

Comment:
So every time this code is run , this error log is printed right ? If the error is none it logs `None` and when the error is `Some` it logs the error. Can we have the log only when the error occurs.